src/c/psychrolib.c

```diff
@@ -53,12 +53,18 @@
  * Global constants
  *****************************************************************************************************/
 
-#define R_DA_IP 53.350    // Universal gas constant for dry air (IP version) in ft∙lbf/lb_da/R
-                          // Reference: ASHRAE Handbook - Fundamentals (2017) ch. 1
-#define R_DA_SI 287.042   // Universal gas constant for dry air (SI version) in J/kg_da/K
-                          // Reference: ASHRAE Handbook - Fundamentals (2017) ch. 1
+#define R_DA_IP 53.350            // Universal gas constant for dry air (IP version) in ft∙lbf/lb_da/R
+                                  // Reference: ASHRAE Handbook - Fundamentals (2017) ch. 1
+#define R_DA_SI 287.042           // Universal gas constant for dry air (SI version) in J/kg_da/K
+                                  // Reference: ASHRAE Handbook - Fundamentals (2017) ch. 1
+#define INVALID -99999            // Invalid value
 
-#define INVALID -99999    // Invalid value
+#define MIN_ITER_COUNT 3          // Minimum number of iterations before exiting while loops.
+
+#define MAX_ITER_COUNT 1000       // Maximum number of iterations before exiting while loops.
```

Comment:
100 or even 20 should be sufficient. 2^20 = 1e+06 so in any bisection that we are doing we should be below the convergence threshold. NR should converge in 3 to 5 iterations given the type of very smooth function we have to work with. What's more important is that if convergence is _not_ reached in MAX_ITER_COUNT then the program should return an error.

Reply:
A couple of points about this I am not sure about:
1) Are you proposing to have different MAX_ITER_COUNT in the two loops?
2) I am not sure about throwing an error if MAX_ITER_COUNT is not reached. This is manly because there may be conditions where this is triggered but you do not necessarily want your program to stop abruptly. How about printing a warning instead?

Reply:
No, one MAX_ITER_COUNT for all loops. 20 should be enough, but put 100 is you want to be extra safe. And if it does not converge in 100 iterations then there is something seriously wrong and we need to have a look, therefore it's appropriate for the program to stop or return -99999. This is just defensive programming - we never expect to get there. If we did at the moment then the program would just hang. At least with a stop or -99999 it won't.

Reply:
OK for MAX_ITER_COUNT but I think that the underlying issue still remains -- for an example of no convergence you can try the following conditions:

``` python
import psychrolib as psy

psy.SetUnitSystem(psy.SI)

print(psy.GetTWetBulbFromRelHum(7, 0.61, 100000))
```

Reply:
I have just looked at this in more details and I think it's now fixed in https://github.com/psychrometrics/psychrolib/pull/24/commits/52232e2a06cc2a836618805f6eca7116d3cbf67b.

The issue was with the the discontinuity of the saturated pressure curve at ~0 deg C (+/-) that leads to the calculation of dew point temperature to oscillate about the zero, thus not converging.  If you assume a range of of +/-0.001, the vapour pressure at 0.001 deg C is 611.26 Pa and 611.10 Pa at -0.001 deg C I simply check if vapour pressure is within that range and break if the condition is met. Note that this will only happen when the loop index reaches the MAX_ITER_COUNT value.
https://github.com/psychrometrics/psychrolib/blob/4573c172f09e76b8cc7ec8cf1fbfb0b0b7911d33/src/python/psychrolib.py#L452-L457

If for some reason it had not converged after that, I throw an exception (MAX_ITER_COUNT set to 100)

https://github.com/psychrometrics/psychrolib/blob/4573c172f09e76b8cc7ec8cf1fbfb0b0b7911d33/src/python/psychrolib.py#L452-L457

I have now implemented this in Python only. If you are happy and there are no additional changes to make, I will propagate the changes to the other languages and add the tests to check this case too so that you can review them.